pkg/provider/azure_vmssflex_cache.go

```diff
@@ -132,7 +168,7 @@ func (fs *FlexScaleSet) getNodeVmssFlexID(nodeName string) (string, error) {
 		return fmt.Sprintf("%v", cachedVmssFlexID), nil
 	}
 
-	cached, err := fs.vmssFlexCache.Get(consts.VmssFlexKey, azcache.CacheReadTypeDefault)
+	cached, err := fs.vmssFlexCache.Get(consts.VmssFlexKey, azcache.CacheReadTypeForceRefresh)
```

Comment:
this means it would call list vmss flex api every time instead of using cache when it's invoked

Reply:
Yes, the reason to do so is because if a new node pool is  just added, then the vmss cache might not contain the info of this scaleset. If a new node is joined into the cluster at this time, refreshing the vms of the existing node pools will not work. 

I agree this means an extra call to list all scalesets. However, comparing with the cost to list VMs in each scaleset (line 179), this cost is trival.

Reply:
Can we get the cache twice? If not found, then we update the cache by calling the API.

Reply:
That also works but might cause more api calls. Let's assume there are 4 existing node pools and 1 new node pool:

The current logic send a call to list all 5 node pools and send another 5 calls to get the nodes in each node pool. So, there will be 1 + 5 = 6 calls in total.

If we get the cache twice, 
-  then in the best case, we need 4 calls to each of the existing node pools if the new node is from one of the existing node pools.
- In the worst case, we need 4 + 1 + 5 = 10 calls if the new node is from the 5th new node pool

Reply:
adding a new node pool is one time operation, while if we use `CacheReadTypeForceRefresh`, it would refresh in every call. Usually, we only set `CacheReadTypeForceRefresh` when cache is missed, you could search `CacheReadTypeForceRefresh` in existing code, that's current logic which could avoid throttling, otherwise vmss cache may not work.

Reply:
I see. That make sense. Thanks for the explanation. I just modified the code to get the cache twice, as suggested by Qi.